lib/widgets/compose_box.dart

```diff
@@ -318,6 +318,44 @@ class _AttachFileButton extends StatelessWidget {
   }
 }
 
+class _AttachMediaButton extends StatelessWidget {
+  const _AttachMediaButton({required this.contentController, required this.contentFocusNode});
+
+  final ContentTextEditingController contentController;
+  final FocusNode contentFocusNode;
+
+  _handlePress(BuildContext context) async {
+    FilePickerResult? result;
+    try {
+      result = await FilePicker.platform.pickFiles(type: FileType.media, allowMultiple: true, withReadStream: true);
+    } catch (e) {
+      // TODO(i18n)
+      showErrorDialog(context: context, title: 'Error', message: e.toString());
+      return;
+    }
+    if (result == null) {
+      return; // User cancelled; do nothing
+    }
+
+    if (context.mounted) {} // https://github.com/dart-lang/linter/issues/4007
+    else {
+      return;
+    }
+
+    final Iterable<_File> files = result.files.map((f) {
+      assert(f.readStream != null);  // We passed `withReadStream: true` to pickFiles.
+      return _File(content: f.readStream!, length: f.size, filename: f.name);
+    });
```

Comment:
Kind of unsatisfying how much of this code gets duplicated between the three buttons, and especially the two that are using FilePicker.

I think it might improve things to factor much of this into something like
```dart
abstract class _AttachUploadsButton extends StatelessWidget {
  // …
  Icon get icon;
  Future<Iterable<_File>?> getFiles(BuildContext context);
  // …
}
```

and then `_AttachMediaButton` and the others would be subclasses just implementing those two items. Those implementations would take care of error dialogs and settings requests.